assets/dev/js/editor/utils/conditions.js

```diff
@@ -32,43 +28,64 @@ Conditions = function() {
 			default:
 				return leftValue === rightValue;
 		}
-	};
+	}
+
+	/**
+	 * Get Condition Value
+	 *
+	 * Retrieves a passed condition's value. Dynamic values take precedence. If there is no dynamic value, this method
+	 * checks for a regular control value.
+	 *
+	 * @since 3.7.0
+	 *
+	 * @param comparisonObject A settings object (e.g. element settings - keys and values)
+	 * @param conditionName The conditioning control's name
+	 * @param subConditionName If the conditioning control's value is an object, and the condition checks for a
+	 * specific property, this is the property name.
+	 * @returns {*}
+	 */
+	getConditionValue( comparisonObject, conditionName, subConditionName ) {
+		let value;
+
+		const dynamicValue = comparisonObject.__dynamic__?.[ conditionName ];
+
+		if ( dynamicValue ) {
+			value = dynamicValue;
+		} else if ( 'object' === typeof comparisonObject[ conditionName ] && subConditionName ) {
+			value = comparisonObject[ conditionName ][ subConditionName ];
+		} else {
+			value = comparisonObject[ conditionName ];
+		}
 
-	this.check = function( conditions, comparisonObject, controls ) {
+		return value;
+	}
+
+	check( conditions, comparisonObject ) {
 		const isOrCondition = 'or' === conditions.relation;
 		let conditionSucceed = ! isOrCondition;
 
-		jQuery.each( conditions.terms, function() {
-			const term = this;
+		conditions.terms.forEach( ( term ) => {
 			let comparisonResult;
 
 			if ( term.terms ) {
-				comparisonResult = self.check( term, comparisonObject, controls );
+				comparisonResult = this.check( term, comparisonObject );
 			} else {
 				// A term consists of a control name to be examined, and a sub key if needed. For example, a term
 				// can look like 'image_overlay[url]' (the 'url' is the sub key). Here we want to isolate the
 				// condition name and the sub key, so later it can be retrieved and examined.
 				const parsedName = term.name.match( /([\w-]+)(?:\[([\w-]+)])?/ ),
 					conditionRealName = parsedName[ 1 ],
-					conditionSubKey = parsedName[ 2 ];
-
-				let value = comparisonObject[ conditionRealName ];
+					conditionSubKey = parsedName[ 2 ],
+					// We use null-safe operator since we're trying to get the current element, which is not always
+					// exists, since it's only created when the specific element appears in the panel.
+					placeholder = elementor.selection.getElements()[ 0 ]
```

Comment:
I have removed any check for placeholder and dynamic values, since those are controls-specific.
There is a controls-specific `check()` method in the new `ControlConditions` class that checks for those things.
Now the `check()` function in `conditions.js` is completely generic.